packages/gopcaml-mode/gopcaml-mode.0.0.1/opam

```diff
@@ -0,0 +1,38 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+version: "0.0.1"
+synopsis:
+  "Ultimate Ocaml editing plugin, providing advanced structural editing, movement and analysis in Emacs"
+description:
+  "Gopcaml-mode is a plugin for emacs for editing OCaml code. It aims to extend existing Emacs editing experience to closer match the features in modern IDEs."
+maintainer: ["kirang@comp.nus.edu.sg"]
+authors: ["Kiran Gopinathan"]
+license: "GPL-3.0-only"
+homepage: "https://gitlab.com/gopiandcode/gopcaml-mode"
+bug-reports: "https://gitlab.com/gopiandcode/gopcaml-mode/issues"
+depends: [
+  "dune" {>= "2.1" & < "2.4"}
```

Comment:
```suggestion
  "ocaml" {>= "4.08.0" & < "4.10.0"}
  "dune" {>= "2.1" & < "2.4"}
```

It fails to build on 4.10
```
#=== ERROR while compiling gopcaml-mode.0.0.1 =================================#
1305# context     2.0.5 | linux/x86_64 | ocaml-variants.4.10.0+rc2 | file:///home/travis/build/ocaml/opam-repository
1306# path        ~/.opam/ocaml-variants.4.10.0+rc2/.opam-switch/build/gopcaml-mode.0.0.1
1307# command     ~/.opam/opam-init/hooks/sandbox.sh build dune build -p gopcaml-mode -j 1 @install @runtest
1308# exit-code   1
1309# env-file    ~/.opam/log/gopcaml-mode-15238-865888.env
1310# output-file ~/.opam/log/gopcaml-mode-15238-865888.out
1311### output ###
1312# [...]
1313# File "ast_zipper.ml", line 653, characters 6-44:
1314# 653 |     | Parsetree.Pmty_functor (loc, o_mt, mt) ->
1315#             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
1316# Error: The constructor Parsetree.Pmty_functor expects 2 argument(s),
1317#        but is applied here to 3 argument(s)
1318#       ocamlc .gopcaml.eobjs/byte/dune__exe__Ast_analysis.{cmi,cmo,cmt} (exit 2)
1319# (cd _build/default && /home/travis/.opam/ocaml-variants.4.10.0+rc2/bin/ocamlc.opt -w -40 -g -bin-annot -I .gopcaml.eobjs/byte -I /home/travis/.opam/ocaml-variants.4.10.0+rc2/lib/async -I /home/travis/.opam/ocaml-variants.4.10.0+rc2/lib/async/async_command -I /home/travis/.opam/ocaml-variants.4.10.0+rc2/lib/async/async_quickcheck -I /home/travis/.opam/ocaml-variants.4.10.0+rc2/lib/async/async_[...]
1320# File "ast_analysis.ml", line 152, characters 4-41:
1321# 152 |   | Parsetree.Pmod_functor (_, omt, mexp) ->
1322#           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
1323# Error: The constructor Parsetree.Pmod_functor expects 2 argument(s),
1324#        but is applied here to 3 argument(s)
1325
1326
1327
1328
```

Reply:
The lower bound is based on my memory, soit maybe too strict

Reply:
Yes - just to be safe I've set the lower bound to 4.09.0